src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -254,287 +241,178 @@ static bool equalMetadataVectors(BSONObj& mdV1, BSONObj& mdV2) {
 
 /* ****************************************************************************
 *
-* bsonCustomMetadataToBson -
+* mergeAttrInfo -
 *
-* Generates the BSON for metadata vector to be inserted in database for a given attribute.
-* If there is no custom metadata, then it returns false (true otherwise).
+* Takes as input the information of a given attribute, both in database (attr) and
+* request (caP), and merged them producing the mergedAttr output. The function returns
+* true if it was an actual update, false otherwise.
 *
 */
-static bool bsonCustomMetadataToBson(BSONObj& newMdV, BSONObj& attr) {
-
-    if (!attr.hasField(ENT_ATTRS_MD)) {
-        return false;
-    }
-
-    BSONArrayBuilder mdNewVBuilder;
-    BSONObj mdV = attr.getField(ENT_ATTRS_MD).embeddedObject();
-
-    for( BSONObj::iterator i = mdV.begin(); i.more(); ) {
-        BSONObj md = i.next().embeddedObject();
-
-        if (md.hasField(ENT_ATTRS_MD_TYPE)) {
-            mdNewVBuilder.append(BSON(ENT_ATTRS_MD_NAME << md.getStringField(ENT_ATTRS_MD_NAME) <<
-                                      ENT_ATTRS_MD_TYPE << md.getStringField(ENT_ATTRS_MD_TYPE) <<
-                                      ENT_ATTRS_MD_VALUE << md.getStringField(ENT_ATTRS_MD_VALUE)));
-         }
-         else {
-            mdNewVBuilder.append(BSON(ENT_ATTRS_MD_NAME << md.getStringField(ENT_ATTRS_MD_NAME) <<
-                                      ENT_ATTRS_MD_VALUE << md.getStringField(ENT_ATTRS_MD_VALUE)));
-         }
-     }
-
-     if (mdNewVBuilder.arrSize() > 0) {
-         newMdV = mdNewVBuilder.arr();
-         return true;
-     }
-     return false;
+static bool mergeAttrInfo(BSONObj& attr, ContextAttribute* caP, BSONObj* mergedAttr)
+{
+  BSONObjBuilder ab;
 
-}
+  /* 1. Add value */
+  valueBson(caP, ab);
 
-/* ****************************************************************************
-*
-* checkAndUpdate -
-*
-* Add the 'attr' attribute to the BSONObjBuilder (which is passed by reference), doing
-* the actual update in the case of ContextAttribute matches.
-*
-* Returns true if the update was done, false if only a "attribute copy" was done. If true,
-* the "actualUpdate" argument (passed by reference) is set to true in the case that the
-* original value of the attribute was different that the one used in the update (this is
-* important for ONCHANGE notifications).
-*/
-static bool checkAndUpdate (BSONObjBuilder& newAttr, BSONObj attr, ContextAttribute ca, bool* actualUpdate) {
+  /* 2. Add type (only if type is not empty in the request, otherwise, we leave type untouched) */
```

Comment:
Dicussed by skype the new wording: "Add type, if present in request. If not, just use the one that is already present in the database"